src/NLog/Targets/WebServiceTarget.cs

```diff
@@ -412,9 +429,18 @@ protected override string GetFormattedContent(string parametersContent)
 
             protected override string GetFormattedParameter(MethodCallParameter parameter, object value)
             {
+                string parameterValue = Convert.ToString(value, CultureInfo.InvariantCulture);
+                string valueString = string.Empty;
+                if (!string.IsNullOrEmpty(parameterValue))
+                {
+                    var sb = new StringBuilder(Math.Max(parameterValue.Length + 20, Target.Parameters.Count > 1 ? 256 : 0));
```

Comment:
The Target.Parameters.Count-check only makes sense when using the same StringBuilder for streaming all parameters.

Unless you optimize the (re-)usage of the StringBuilder, then it should just be:

var sb = new StringBuilder(parameterValue.Length + 20);